app/src/main/java/com/android/sample/model/userAccount/UserAccountRepositoryFirestore.kt

```diff
@@ -64,4 +64,19 @@ class UserAccountRepositoryFirestore(private val db: FirebaseFirestore) : UserAc
           onFailure(exception)
         }
   }
+
+  override fun deleteUserAccount(
+      userId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .document(userId)
+        .delete()
+        .addOnSuccessListener { onSuccess() }
+        .addOnFailureListener { exception ->
+          Log.e("UserAccountRepo", "Error deleting user account", exception)
+          onFailure(exception)
+        }
+  }
```

Comment:
If a workout is not shared among multiple users, the workouts linked to the user should also be deleted (Or the user should be removed from the UserSet of the workout he was on).
But maybe It should be done in an other PR since your PR came before the shared workout feature. 
Tell me what you think.

Reply:
Yes it's only a line of code so I think adding it to a next workout related PR is feasible but thanks for the suggestion you're totally right